src/cirq_iqm/iqm_client.py

```diff
@@ -127,15 +127,14 @@ def submit_circuit(
         Returns:
             ID for the created task. This ID is needed to query the status and the execution results.
         """
-        if qubit_mapping is None:
-            qubit_mapping = []
 
         data = {
-            "qubit_mapping": qubit_mapping,
+            "qubit_mapping": [q.dict() for q in qubit_mapping],
             "circuit": circuit.dict(),
             "settings": self._settings,
             "shots": shots
         }
```

Comment:
Is there a reason why the entire data transfer object we send isn't a single Pydantic model, containing various submodels?

Reply:
I don't see any reason not to so I added that :)